src/include/settings/settings_defs.h

```diff
@@ -0,0 +1,212 @@
+
+// Enable or disable brain
+SETTING_bool(brain, "Enable brain (default: false)", false, true, terrier::DBMain::EmptyCallback)
+
+/*
+SETTING_string(terrier_address,
+"ip and port of the terrier rpc service, address:port",
+"127.0.0.1:15445",
+false, false)
+*/
+
+// Size of the brain task queue
+SETTING_int(brain_task_queue_size, "Brain Task Queue Size (default: 32)", 32, 1, 128, false, terrier::DBMain::EmptyCallback)
+
+// Size of the brain worker pool
+SETTING_int(brain_worker_pool_size, "Brain Worker Pool Size (default: 1)", 1, 1, 16, false, terrier::DBMain::EmptyCallback)
+
+//===----------------------------------------------------------------------===//
+// CODEGEN
+//===----------------------------------------------------------------------===//
+
+// If code-generation is enabled
+SETTING_bool(codegen, "Enable code-generation for query execution (default: true)", true, true,
+             terrier::DBMain::EmptyCallback)
+
+// If llvm code is interpreted
+SETTING_bool(codegen_interpreter, "Force interpretation of generated llvm code (default: false)", false, true,
+             terrier::DBMain::EmptyCallback)
+
+// If statics will be printed on generated IR
+SETTING_bool(print_ir_stats, "Print statistics on generated IR (default: false)", false, true,
+             terrier::DBMain::EmptyCallback)
+
+// If logging of all generated IR is enabled
+SETTING_bool(dump_ir, "Enable logging of all generated IR (default: false)", false, true, terrier::DBMain::EmptyCallback)
+
+//===----------------------------------------------------------------------===//
+// Optimizer
+//===----------------------------------------------------------------------===//
+SETTING_bool(predicate_push_down, "Enable predicate push-down optimization (default: true)", true, true,
+             terrier::DBMain::EmptyCallback)
+
+// If bloom filter for hash join in codegen is enabled
+SETTING_bool(hash_join_bloom_filter, "Enable bloom filter for hash join in codegen (default: false)", false, true,
+             terrier::DBMain::EmptyCallback)
+
+// Timeout for task execution
+SETTING_int(task_execution_timeout,                             // NOLINT
+            "Maximum allowed length of time (in ms) for task "  // NOLINT
+            "execution step of optimizer, "                     // NOLINT
+            "assuming one plan has been found (default 5000)",  // NOLINT
+            5000, 1000, 60000, true, terrier::DBMain::EmptyCallback)     // NOLINT
+// NOLINT
+
+//===----------------------------------------------------------------------===//
+// GENERAL
+//===----------------------------------------------------------------------===//
+
+// Display configuration
+SETTING_bool(display_settings, "Display settings (default: false)", false, true, terrier::DBMain::EmptyCallback)
+
+//===----------------------------------------------------------------------===//
+// Used only in Tests
+//===----------------------------------------------------------------------===//
+
```

Comment:
Having test values in the main code is not too nice. 
These should be in the test directory with whatever test is exercising these values.